utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/azure/SchemaNameTopicMatchStrategy.scala

```diff
@@ -1,60 +1,96 @@
+  )
 
   def keySchemaNameFromTopicName(topicName: String): String = schemaNameFromTopicName(topicName, isKey = true)
 
   def valueSchemaNameFromTopicName(topicName: String): String = schemaNameFromTopicName(topicName, isKey = false)
 
   def schemaNameFromTopicName(topicName: String, isKey: Boolean): String = {
     val suffix = if (isKey) KeySuffix else ValueSuffix
-    CaseFormat.LOWER_HYPHEN.to(CaseFormat.UPPER_CAMEL, topicName) + suffix
+    toSchemaNamingConvention(topicName) + suffix
   }
 
+  /**
+    * Transforms topic name to schema naming convention.
+    * @see https://nussknacker.io/documentation/docs/integration/KafkaIntegration/#association-between-schema-with-topic
+    */
+  def toSchemaNamingConvention(topicName: String): String =
+    topicName.toLowerCase.replaceAll("\\W", " ").split(" ").map(_.capitalize).mkString("")
+
   object FullSchemaNameDecomposed {
 
-    private val fullyQualifiedRegex = "^(.*)\\.[^.]*$".r
-
-    // Decompose schema (full) name to: topicName, namespace and isKey
-    def unapply(schemaFullName: String): Option[(String, Option[String], Boolean)] = {
-      // Only schemas ends with Key or Value will be recognized - we assume that other schemas are for other purpose
-      // and won't appear on any Nussknacker lists.
-      if (schemaFullName.endsWith(KeySuffix)) {
-        val (topicName, namespace) = extractTopicNameAndNamespace(schemaFullName, KeySuffix)
-        Some(topicName, namespace, true)
-      } else if (schemaFullName.endsWith(ValueSuffix)) {
-        val (topicName, namespace) = extractTopicNameAndNamespace(schemaFullName, ValueSuffix)
-        Some(topicName, namespace, false)
-      } else {
-        None
-      }
-    }
+    private def fullSchemaNamePattern(suffix: String) = ("^(.*\\.)?([^.]*)" + suffix + "$").r
+    private val namespaceAndNameKey                   = fullSchemaNamePattern(KeySuffix)
+    private val namespaceAndNameValue                 = fullSchemaNamePattern(ValueSuffix)
 
-    private def extractTopicNameAndNamespace(schemaFullName: String, suffix: String): (String, Option[String]) = {
-      val schemaFullNameWithoutSuffix = schemaFullName.replaceFirst(suffix + "$", "")
-      // We remove namespace from topic name because we don't want to force ppl to create technically looking event hub names
-      // Also event hubs already has dedicated namespaces
-      val cleanedSchemaName = schemaFullNameWithoutSuffix.replaceFirst("^.*\\.", "")
-      // Topics (event hubs) can have capital letters but generally it is a good rule to not use them in topic names to avoid
-      // mistakes in interpretation shortcuts and so on - similar like with sql tables
-      val topicName = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_HYPHEN, cleanedSchemaName)
-      val namespace = fullyQualifiedRegex.findFirstMatchIn(schemaFullNameWithoutSuffix).map { m =>
-        m.group(1)
+    /**
+      * Decompose schema (full) name to: topicName, namespace (optional) and isKey.
```

Comment:
Is itn't relevant anymore, it decomposes schema to "core" part of the name.

Reply:
fixed description